core/src/main/java/nl/nn/adapterframework/doc/IbisDocPipe.java

```diff
@@ -384,96 +385,153 @@ public PipeRunResult doPipe(Object input, IPipeLineSession session) throws PipeR
 		return new PipeRunResult(getForward(), result);
 	}
 
-	private String getJson() {
-		Map<String, TreeSet<IbisBean>> groups = getGroups();
-		IbisDocExtractor extractor = new IbisDocExtractor();
-
-		// For all folders
-		for (String folder : groups.keySet()) {
-
-			// For all classes
-			for (IbisBean ibisBean : groups.get(folder)) {
-
-				// Copy the properties of FileSender into FilePipe so that the properties of FileHandler are also in FilePipe
-				Map<String, Method> beanProperties = getBeanProperties(ibisBean.getClazz());
-				if (beanProperties != null) {
-
-					if (ibisBean.getName().equals("FilePipe")) {
-						for (IbisBean bean : groups.get("Senders")) {
-							if (bean.getName().equals("FileSender")) {
-								Map<String, Method> senderProperties = getBeanProperties(bean.getClazz());
-								if (senderProperties != null) {
-									beanProperties.putAll(senderProperties);
-								}
-								break;
-							}
-						}
-					}
-
-					// For each method in the class
-					Iterator<String> iterator = new TreeSet<>(beanProperties.keySet()).iterator();
-
-					while (iterator.hasNext()) {
-
-						// Get the method
-						String property = iterator.next();
-						Method method = beanProperties.get(property);
-
-						// Get the ibisdoc of the method
-						IbisDoc ibisDoc = AnnotationUtils.findAnnotation(method, IbisDoc.class);
+	/**
+	 * Get the superclasses of a certain class.
+	 *
+	 * @param clazz - The class we have to derive the superclasses from
+	 * @return An ArrayList containing all the superclasses with priority given to them
+	 */
+	private ArrayList<String> getSuperClasses(Class clazz) {
+		ArrayList<String> superClasses = new ArrayList<String>();
+		int index = 0;
+		while (clazz.getSuperclass() != null) {
+
+			// Assign a string with a priority number attached to it and add it to the array of superclasses
+			String str = Integer.toString(index);
+			superClasses.add(clazz.getSimpleName() + str);
```

Comment:
Waarom voeg je de index aan de naam toe? De lijst is toch op dezelfde volgorde als de index?

Reply:
Klopt, ik heb het in eerste instantie op een inefficiente manier gedaan maar dat is nu gefixt.